api/authenticator.py

```diff
@@ -465,7 +465,7 @@ def authenticated_patron(
             ProblemDetail if an error occurs.
         """
         provider: AuthenticationProvider | None = None
-        provider_token: Dict[str, str] | str | None = None
+        provider_token: Dict[str, str | None] | str | None = None
```

Comment:
Werkzeug modified their type hints to allow a None to be returned, so we also need to update our hints.